src/metrics/timingMetric.js

```diff
@@ -0,0 +1,43 @@
+import {VALID_TYPES} from './metrics';
+
+function typeOrError(type) {
+  if (VALID_TYPES.includes(type)) {
+    return type;
+  }
+  throw new Error('invalid type for a metric supplied');
+}
+
+export default function newMetric(metricType, {fields, tags} = {
+  fields: [],
+  tags: []
+}, stop = undefined, start = undefined) {
+  const type = typeOrError(metricType);
+  const startTime = start || new Date();
+
+  return {
+    addField(key, value) {
+      fields[key] = value;
+      return this;
+    },
+    addTag(key, value) {
+      tags[key] = value;
+      return this;
+    },
+    get() {
```

Comment:
Personally I'd go with `.getMeasurement()`. `.get()` feels a bit OO to me and doesn't describe what it's returning here.